Git-Workflow-Guidance.md

```diff
@@ -414,62 +479,81 @@ If all you want to do is change the Icepack version in CICE in the repository, t
+      cd CICE.icepack.feature
+      mv Icepack Icepack.00                            # move the default Icepack
+      git clone https://github.com/username/Icepack    # clone Icepack separately
+      cd Icepack
+      git branch feature
+      git checkout feature
 
-1.  clone CICE from user fork
-1.  update CICE fork master to Consortium version
-1.  create and checkout a CICE fork branch
-1.  clone Icepack to user fork
-1.  update Icepack fork master to Consortium version
-1.  create and checkout Icepack fork branch
-1.  develop/commit/push Icepack changes to Icepack fork branch, develop/commit/push CICE changes to CICE fork branch, never update CICE submodule formally in fork branch
-1.  execute PR for Icepack to merge changes to Consortium Icepack master
-1.  update CICE fork branch submodule to point to the appropriate Consortium Icepack version and commit/push that change to the CICE fork branch
-1.  execute PR for CICE to merge changes to Consortium CICE master including updating the pointer to the Icepack version
+You now have a copy of Icepack from your personal fork on a new branch in your CICE sandbox.  You can develop, test, commit, and push Icepack changes.  You can also test CICE in this sandbox.  Eventually, you can create an Icepack Pull Request.  In this mode of development, you will probably not be committing any changes to the CICE repository nor should you worry about anything related to the submodule in CICE.  You are just developing Icepack within a CICE sandbox.
 
-By using this approach, only at the final CICE commit is the Icepack submodule updated.  There may be times where this is not adequate, but as a default, it’s probably the easiest way to carry out development of Icepack and CICE in parallel.  More specifically, this might look something like
+## Develop CICE and Icepack Concurrently
+
+There are a number of different ways to handle development of Icepack and CICE concurrently.  What we strongly recommend is that you checkout CICE and Icepack from your fork and develop each within a single sandbox but as independent repositories.  Eventually, you'll do a PR for Icepack.  Once that is accepted, you can [update the submodule in CICE](https://github.com/CICE-Consortium/About-Us/wiki/Git-Workflow-Guidance/#Update-Icepack-in-CICE) and create a CICE PR.  The basic steps are
+
+1.  clone CICE from user repository (fork)
+1.  update CICE repository master to Consortium version if needed
+1.  create and checkout a CICE branch
+1.  remove/move Icepack
+1.  clone Icepack from user repository (fork)
+1.  update Icepack repository master to Consortium version if needed
+1.  create and checkout an Icepack branch
+1.  develop/commit/push Icepack changes to the Icepack remote repository on the Icepack branch.  Develop/commit/push CICE changes to CICE remote repository on the CICE branch.  **Do not update the Icepack submodule in CICE**
```

Comment:
Also 
> **Do not update the Icepack submodule in CICE**

this could be more precise ; are we talking about A) (in CICE)
~~~bash
git add icepack
git commit
~~~
or B) (also in CICE)
~~~bash
git submodule update
~~~
? 

If we don't commit in CICE as we go along, we are not doing A), and we should not either do B) since this will checkout Icepack at the commit recorded in CICE and detach Icepack's HEAD, and then we would have to 
~~~bash
cd icepack
git checkout feature
~~~
(this should be mentioned if we stay with this workflow)

If however we do commit in Icepack and CICE as we go (we do A)), then doing B) is not harmful because in this case `git submodule udpate` [will *not* detach HEAD](https://git-scm.com/docs/git-submodule#Documentation/git-submodule.txt-checkout), so we do not have to go back in Icepack and switch branches again.

Reply:
The submodule thing is creating problems even for our group of developers.  What I have recommended is to work on Icepack and CICE separately and if you're working with others, to communicate.  Changing the fork of the submodule is not something we want to ever recommend at this point, but that's what would be needed to collaborate with compliant versions of the submodule.  I really don't want to add this level of complexity to our documentation.  Again, I want to simplify where I can and treat this as a getting started guide.  What we recommend should work and it should not introduce too many problems.  I agree that it's not sophisticated, but more experienced users won't really reference this guide.  It's the beginners we need to target.

In the above I think I was thinking of A.  Separately in the document, I think I note not to do B if you have a modified version of Icepack in your sandbox.  Generally, both A and B should be avoided if a user is doing anything with Icepack under CICE.  We can make that clearer.

Reply:
> Changing the fork of the submodule is not something we want to ever recommend at this point, but that's what would be needed to collaborate with compliant versions of the submodule.

As I showed [here](https://github.com/phil-blain/CICE-Consortium-About-Us-wiki/pull/1#discussion_r358936302) , it is not necessary to "change the fork" (I think here you mean changing the Icepack URL in `.gitmodules`) in order for a local checkout of CICE to reference a different Icepack fork than the Consortium repo, it can be changed in the local configuration only. But I agree it's more involved on the technical level. 

I can live with not commiting in CICE.

> Generally, both A and B should be avoided if a user is doing anything with Icepack under CICE. We can make that clearer.

If we go this way, yes. We indeed should make that clearer.